kernel-util/src/main/java/org/sakaiproject/content/util/ZipContentUtil.java

```diff
@@ -394,8 +413,34 @@ private void storeContentCollection(String rootId, ContentCollection collection,
 	 * @param out
 	 * @throws Exception
 	 */
-	private void storeContentResource(String rootId, ContentResource resource, ZipOutputStream out) throws Exception {		
-		String filename = resource.getId().substring(rootId.length(),resource.getId().length());				
+	private void storeContentResource(String rootId, ContentResource resource, ZipOutputStream out) throws Exception {
+		String filename = resource.getId().substring(rootId.length(),resource.getId().length());
+
+		//Inorder to have username as the folder name rather than having eids
+		if(rootId.indexOf("group-user")!=-1 && ServerConfigurationService.getBoolean("dropbox.zip.haveDisplayname", true)) {
+			if (filename != null && filename.length() > 0) {
+				try {
+					String filenameArr[] = filename.split(Entity.SEPARATOR);
+					ContentCollectionEdit collectionEdit;
+					ResourcePropertiesEdit props;
+					String displayName;
+					String uniqueId;
+					if (filenameArr.length > 1) {
+						collectionEdit = (ContentCollectionEdit) ContentHostingService.getCollection(rootId + filenameArr[0] + Entity.SEPARATOR);
+						props = collectionEdit.getPropertiesEdit();
+						displayName = props.getProperty(ResourcePropertiesEdit.PROP_DISPLAY_NAME);
+						filename = displayName + Entity.SEPARATOR + filenameArr[1];
+					} else {
+						collectionEdit = (ContentCollectionEdit) ContentHostingService.getCollection(rootId);
+						props = collectionEdit.getPropertiesEdit();
+						displayName = props.getProperty(ResourcePropertiesEdit.PROP_DISPLAY_NAME);
+						filename = displayName + Entity.SEPARATOR + filenameArr[0];
+					}
+				} catch (Exception e) {
+					LOG.warn("Unexpected error occurred when trying to create Zip archive:" + extractName(rootId));
```

Comment:
If you're logging a warn and it's not expected in normal operations, it's useful to include the stack trace as it makes debugging the problem much easier.